src/lua/httpc.lua

```diff
@@ -262,13 +264,38 @@ curl_mt = {
             if not method or not url then
                 error('request(method, url[, body, [options]])')
             end
-            local resp = self.curl:request(method, url, body, opts or {})
+            opts = opts or {}
+            opts.params = opts.params or {}
+            local query_encoder = require("uri")._internal.params
+            local url_with_params = url
+            local ok, res = pcall(query_encoder, opts.params)
+            if (ok == false) then
+                error(res)
+            end
+            local encoded_params = res
+            if encoded_params then
+                if method == "GET" or
+                   method == "HEAD" or
+                   method == "DELETE" then
```

Comment:
Why are the methods in uppercase? In the other patch you mentioned that they should be in lover:
https://www.rfc-editor.org/rfc/rfc9110.html#section-5.6.6